### DISCUSSION TITLE
Does Keystore wallet worked for interacting a contract using Cast Send

### DISCUSSION THREAD

**Discussion by ALtera21** [Cyfrin/foundry-full-course-cu#2705] [Q&A] [answered]
Does Keystore supposed to work for cast send and cast call ?
I tried to interact with a SimpleStorage contract with these command in bash:
```solidity
cast send 0x5FbDB2315678afecb367f032d93F642f64180aa3 "store(uint256)" 112 --rpc-url http://127.0.0.1:8545 --account DefaultAnvil --sender 0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266
```

and somehow it didn't recognize  --sender and output this error:
```bash
error: unexpected argument '--sender' found

  tip: a similar argument exists: '--ledger'

Usage: cast call --account <ACCOUNT_NAME> --ledger <--rpc-url <URL>|--flashbots|--jwt-secret <JWT_SECRET>> <TO> <SIG> [ARGS]...

For more information, try '--help'.
```
 
It does worked when i used `--private-key`
and also, the keystore only worked when i used it to deploy a contract using scripts and didn't worked when using `forge create`, using forge create had the same result as the error mentioned above, is this normal?

**Comment by ALtera21** [comment, marked as answer]
I am s stupid, the `--sender` caused the error and i don't even think of just removing it.

**Reply by EngrPips** [reply]
Interesting, glad that you figured that out.

**Reply by satyaorz** [reply]
i am facing the same issue, but i have a doubt. Like if we don't put ```--sender``` how will the contract know, which address the fund came from? sending fund and using the store function will cause a transaction too right? 
edit: oh nevermind, its the privatekey thats required only, so it will do the job i guess?

**Reply by EngrPips** [reply]
Indeed
